test/controllers/fhir/r4/patient_test.rb

```diff
@@ -474,6 +503,206 @@ class ApiControllerTest < ActionDispatch::IntegrationTest
     assert_equal false, json_response['active']
   end
 
+  test 'should update Patient follow up reason via update' do
+    Patient.find_by(id: 1).update!(
+      follow_up_reason: 'High-Risk'
+    )
+
+    put(
+      '/fhir/r4/Patient/1',
+      params: Patient.find_by(id: 1).as_fhir.to_json,
+      headers: { Authorization: "Bearer #{@system_patient_token_rw.token}", 'Content-Type': 'application/fhir+json' }
+    )
+    assert_response :ok
+    json_response = JSON.parse(response.body)
+    assert_equal 1, json_response['id']
+    p = Patient.find_by(id: 1)
+
+    assert_not p.nil?
+    assert_equal 'Patient', json_response['resourceType']
+    assert_equal p.follow_up_reason, fhir_ext_str(json_response, 'follow-up-reason')
+    assert_equal 'High-Risk', p.follow_up_reason
+  end
+
+  test 'should not be able to update Patient follow up reason to invalid reason' do
+    Patient.find_by(id: 1).update!(
+      follow_up_reason: 'Some invalid reason'
+    )
+
+    put(
+      '/fhir/r4/Patient/1',
+      params: Patient.find_by(id: 1).as_fhir.to_json,
+      headers: { Authorization: "Bearer #{@system_patient_token_rw.token}", 'Content-Type': 'application/fhir+json' }
+    )
+    assert_response :unprocessable_entity
+    json_response = JSON.parse(response.body)
+    issues = json_response['issue']
+
+    assert_equal 1, issues.length
+    assert(json_response['issue'][0]['diagnostics'].include?('Value \'Some invalid reason\' for \'Follow-up Reason\' is not an acceptable value'))
+  end
+
+  test 'should update Patient follow up note via update' do
+    Patient.find_by(id: 1).update!(
+      follow_up_note: 'New follow up note'
+    )
+
+    put(
+      '/fhir/r4/Patient/1',
+      params: Patient.find_by(id: 1).as_fhir.to_json,
+      headers: { Authorization: "Bearer #{@system_patient_token_rw.token}", 'Content-Type': 'application/fhir+json' }
+    )
+    assert_response :ok
+    json_response = JSON.parse(response.body)
+    assert_equal 1, json_response['id']
+    p = Patient.find_by(id: 1)
+
+    assert_not p.nil?
+    assert_equal 'Patient', json_response['resourceType']
+    assert_equal p.follow_up_note, fhir_ext_str(json_response, 'follow-up-note')
+    assert_equal 'New follow up note', p.follow_up_note
+  end
+
+  test 'should remove Patient follow up note via update' do
```

Comment:
Although it feels wrong to say this...I think this test may be too much testing. Since we know that updating the value of `follow_up_note` works from other tests, and we test that `nil` is one of the allowed values elsewhere, I don't think we need to test this here. But then again, if you want to ignore me, I'm not going to double down against tests.

Reply:
I don't think there is a test that `nil` is one of the allowed values. Should I add an assertion in one of the other tests where `follow_up_note` is nil?

Reply:
I think since this test: https://github.com/SaraAlert/SaraAlert/blob/73ca4747e18d2d0cf1d9c694b9ea5dbbd5113ba6/test/models/patient_test.rb#L2011 checks that we are allowed to set `follow_up_note` to `nil`, and we already have a test in this file which tests we can set `follow_up_note`, those two tests combine to make me feel confident that setting `follow_up_note` to `nil` will work.